packages/edit-site/src/components/sidebar-navigation-screen-pattern/template-part-navigation-menu-list-item.js

```diff
@@ -1,17 +1,17 @@
 /**
  * WordPress dependencies
  */
-import { useEntityProp } from '@wordpress/core-data';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import SidebarNavigationItem from '../sidebar-navigation-item';
 import { useLink } from '../routes/link';
+import useEditedEntityRecord from '../use-edited-entity-record';
 
 export default function TemplatePartNavigationMenuListItem( { id } ) {
-	const [ title ] = useEntityProp( 'postType', 'wp_navigation', 'title', id );
+	const { getTitle } = useEditedEntityRecord( 'wp_navigation', id );
```

Comment:
Seems like a good solution for now.

I wonder if in the future we should have a selector for `getEntityTitle( id )` which does this? It's so common and it causes so many bugs all the time.

I think you should propose it.

Reply:
This doesn't do anything different internally. See https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/use-edited-entity-record/index.js#L59.

I think we just need to make sure entities return values like titles in a predictable way.

Reply:
Oh really. Got to be honest I was doing a fly by review. My comment probably wasn't super helpful.

My thought was we need a single source of truth for getting a string-based title. I think the `raw` vs `rendered` thing throws people. Maybe if we had some well-named utilities that got the correct title based on where you want to output it that night help.

`getEntityTitleForField` vs `getEntityTitle` or something. Just an idea...

Reply:
I updated this to just use `getEditedEntityRecord`. I'm not sure what the best way to do this is, but right now we have a JS error in trunk and this branch fixes it, so I'd rather get this merged and iterate on improvements later.

Reply:
I’ll be able to do another round of reviews tomorrow morning.

Can you post more details on how to reproduce issue issue you’re having on trunk?

Reply:
I can't get it to happen since https://github.com/WordPress/gutenberg/pull/52707 merged. Still investigating...